presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java

```diff
@@ -399,7 +422,74 @@ private static Signature bindSignature(Signature signature, List<? extends Type>
                 boundArguments.add(lastArgument);
             }
         }
-        return new Signature(signature.getName(), signature.getType(), bindParameters(signature.getReturnType(), boundParameters), boundArguments.build());
+        Signature boundSignature = new Signature(signature.getName(), signature.getType(), bindParameters(signature.getReturnType(), boundParameters), boundArguments.build());
+        return boundSignature;
+    }
+
+    private static boolean containsArgumentsWithLiteralParameters(Signature signature)
+    {
+        for (TypeSignature typeSignature : signature.getArgumentTypes()) {
+            if (!typeSignature.getLiteralParameters().isEmpty()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static class SignatureWithExtractedTypesWithLiteralParameters
```

Comment:
maybe move this class outside with related methods (containsArgumentsWithLiteralParameters, extractTypeParametersForArgumentsWithLiteralParameters).